fda-model/structures/Inventory.cs

```diff
@@ -95,148 +109,136 @@ public static T TryGetObj<T>(object value, T defaultValue = default)
                 return defaultValue;
         }
     }
+    #endregion
 
+    #region Methods
     public Polygon GetImpactAreaPolygon(string impactAreaName)
     {
-        for (int i = 0; i < _impactAreaSet.FeatureCount(); i++)
+        PolygonFeatureLayer impactAreas = new PolygonFeatureLayer("ImpactAreas", _impactAreaShapefile);
+        for (int i = 0; i < impactAreas.FeatureCount(); i++)
         {
-            var row = _impactAreaSet.FeatureRow(i);
+            var row = impactAreas.FeatureRow(i);
             string thisImpactAreaName = TryGetObj<string>(row[_impactAreaUniqueColumnHeader]);
-            if (thisImpactAreaName.Equals(impactAreaName));
+            if (thisImpactAreaName.Equals(impactAreaName)) ;
             {
-                return _impactAreaSet.Polygon(i);
+                return impactAreas.Polygon(i);
             }
         }
         return null;
     }
-
-    private PointFeatureLayer createColumnHeadersForMissingColumns(PointFeatureLayer layer, StructureInventoryColumnMap map)
+    private void LoadStructuresFromSourceFiles()
     {
-        List<string> layerColumnNames = new List<string>();
-        var row = layer.FeatureRow(0);
-        foreach (DataColumn c in row.Table.Columns)  //loop through the columns. 
+        PointFeatureLayer structureInventory = new PointFeatureLayer("Structure_Inventory", _structureInventoryShapefile);
+        createColumnHeadersForMissingColumns(ref structureInventory, _map);
+        PolygonFeatureLayer impactAreaSet = new PolygonFeatureLayer("Impact_Area_Set", _impactAreaShapefile);
+
+        float[] groundelevs = Array.Empty<float>();
+        if (_updateGroundElevsFromTerrain)
         {
-            layerColumnNames.Add(c.ColumnName);
+            groundelevs = GetGroundElevationFromTerrain(_structureInventoryShapefile, _terrainPath);
         }
 
-        foreach (Tuple<string,Type> nameTypePair in map.ColumnHeaders)
+        PointMs pointMs = new PointMs(structureInventory.Points().Select(p => p.PointM()));
+        for (int i = 0; i < structureInventory.FeatureCount(); i++)
         {
-            if (!layerColumnNames.Contains(nameTypePair.Item1))
+            //required parameters
+            PointM point = pointMs[i];
+            var row = structureInventory.FeatureRow(i);
+            int fid = TryGet<int>(row[_map.StructureID], -999);
```

Comment:
I wonder if we should have a static constant -999 somewhere we can reference  everywhere for our null values so we don't have all of this hard coded stuff.